### PR TITLE
add consul min tls version

### PR DESCRIPTION
**Pull request by ashwinexe**
## Related Issue(s)

 #446 
## Description
We want to create a policy for Consul service mesh to add a policy for the mesh configuration. The policy will check the TLS Min version to ensure that whenever the mesh is configured, there is a minimum version of TLS set for all the service mesh proxies.
cc @shivaylamba

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by ashwinexe** [comment]
@chipzoller our test cases are failing can you take a look?

After following the [test dcoumentation](https://kyverno.io/docs/kyverno-cli/#test). 


image

PS: The test passed on `skip` for some reason.
The policy is working when applied independently on our console service mesh with TLS version 1.2 (pass) and 1.1 (fail)

File: consul/enforce-min-tls-version/kyverno-test.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: enforce-min-tls-version
+policies:
+  - enforce-min-tls-version.yaml
+resources:
+  - resource.yaml
+results:
+##### Pods - Bad
+  - policy: enforce-min-tls-version
+    rule: adding-capabilities
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/kyverno-test.yaml:9]: rule name reference is incorrect

File: consul/enforce-min-tls-version/kyverno-test.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    resource: badmesh
+    kind: Mesh
+    result: fail
+  ###### Pods - Good
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/kyverno-test.yaml:13]: Please remove comments. They are not applicable.

File: consul/enforce-min-tls-version/kyverno-test.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - policy: enforce-min-tls-version
+    rule: adding-capabilities
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/kyverno-test.yaml:15]: rule name reference is incorrect

File: consul/enforce-min-tls-version/resource.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+###### Pods - Bad
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/resource.yaml:1]: Please remove comments. They are not applicable.

File: consul/enforce-min-tls-version/resource.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+apiVersion: consul.hashicorp.com/v1alpha1
+kind: Mesh
+metadata:
+  name: badmesh
+spec:
+  tls:
+    incoming:
+      tlsMinVersion: TLSv1_1
+--- 
+###### Pods - Good
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/resource.yaml:11]: Please remove comments. They are not applicable.

File: consul/enforce-min-tls-version/kyverno-test.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+name: enforce-min-tls-version
+policies:
+  - enforce-min-tls-version.yaml
+resources:
+  - resource.yaml
+results:
+##### Pods - Bad
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/kyverno-test.yaml:7]: Please remove comments. They are not applicable.

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: consul/enforce-min-tls-version/kyverno-test.yaml (hunk 1 above)
Comment (ashwinexe) [reply consul/enforce-min-tls-version/kyverno-test.yaml:9]: fixed: rule updated to `check-for-tls-version`

**Comment by ashwinexe** [comment]
Hey @chipzoller thanks, it worked. All tests are passing now.
image

File: consul/enforce-min-tls-version/enforce-min-tls-version.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+ name: enforce-min-tls-version
+ annotations:
+   policies.kyverno.io/title: Enforce Consul min TLS version 
+   policies.kyverno.io/category: Consul
+   policies.kyverno.io/severity: medium
+   policies.kyverno.io/subject: TlsVersion
```
Comment (chipzoller) [inline consul/enforce-min-tls-version/enforce-min-tls-version.yaml:9]: Subject here is "Mesh" and not TlsVersion.

**Review by chipzoller — CHANGES_REQUESTED** [review]
You must push changes prior to resolving comments.

**Comment by chipzoller** [comment]
Please also avoid posting screenshots of text in issues and PRs. This is because it cannot be searched and we may need to locate such things in the future. Please use code blocks and copy-and-paste text as much as possible.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**